how-to/common/public/windows/intents/instance-picker.js

```diff
@@ -0,0 +1,253 @@
+import { createOptionEntry, setElementVisibility } from './helper.js';
+
+let rejectAppSelection;
+let resolveAppSelection;
+let launchBtn;
+let cancelSelectionBtn;
+let intentsContainer;
+let appsContainer;
+let intentSelectionContainer;
+let appSelectionContainer;
+let appInstanceSelectionContainer;
+let appInstanceContainer;
+let appSummaryContainer;
+let targetLabel;
+let targetLabelSet = false;
+
+let intent;
+let intents;
+let apps;
+const appLookup = {};
+
+async function setupIntentView(setupIntents) {
+	if (Array.isArray(setupIntents) && setupIntents.length > 0) {
+		if (intent.context?.type !== undefined) {
+			targetLabel.textContent = `Select an Intent & Application to handle the ${intent.context.type} context: ${intent.context.name}`;
+			targetLabelSet = true;
+		}
+
+		for (let i = 0; i < setupIntents.length; i++) {
+			const intentEntry = createOptionEntry(
+				setupIntents[i].intent.displayName,
+				setupIntents[i].intent.name,
+				i === 0
+			);
+			intentsContainer.append(intentEntry);
+		}
+
+		setElementVisibility(appSelectionContainer, false);
+		setElementVisibility(appInstanceSelectionContainer, false);
+		setElementVisibility(intentSelectionContainer, true);
+		await onIntentSelection(setupIntents[0].intent.name);
+	} else {
+		setElementVisibility(intentSelectionContainer, false);
+	}
+}
+
+async function setupAppView(applications, intentName) {
+	setElementVisibility(appInstanceSelectionContainer, false);
+	setElementVisibility(appSummaryContainer, false);
+	appsContainer.options.length = 0;
+	if (Array.isArray(applications) && applications.length > 0) {
+		if (intentName !== undefined && !targetLabelSet) {
+			targetLabel.textContent = `Select an application to handle the Intent ${intentName}`;
+		}
+		for (let i = 0; i < applications.length; i++) {
+			if (applications[i].instanceId === undefined) {
+				const appEntry = createOptionEntry(applications[i].title, applications[i].appId, i === 0);
+				appsContainer.append(appEntry);
+			}
+		}
+		setElementVisibility(appSelectionContainer, true);
+		await onAppSelection(applications[0].appId);
+	} else {
+		setElementVisibility(appSelectionContainer, false);
+	}
+}
+
+async function setupAppInstancesView(foundAppInstances) {
+	setElementVisibility(appInstanceSelectionContainer, false);
+	appInstanceContainer.options.length = 0;
+	launchBtn.disabled = false;
+	if (Array.isArray(foundAppInstances) && foundAppInstances.length > 0) {
+		const newInstanceEntry = createOptionEntry('New Instance', '', true);
+		appInstanceContainer.append(newInstanceEntry);
+		for (let i = 0; i < foundAppInstances.length; i++) {
+			if (foundAppInstances[i].instanceId !== undefined) {
+				const appMetadata = await fdc3.getAppMetadata(foundAppInstances[i]);
+				let label = `${foundAppInstances[i].appId} (${i + 1})`;
+				if (
+					appMetadata?.instanceMetadata !== undefined &&
```

Comment:
`appMetadata?.instanceMetadata !== undefined &&` is not needed in this test

Reply:
updated